client-spark/common/src/main/java/org/apache/uniffle/shuffle/manager/ShuffleManagerGrpcService.java

```diff
@@ -101,6 +104,48 @@ public void reportShuffleFetchFailure(
     responseObserver.onCompleted();
   }
 
+  @Override
+  public void getPartitionToShufflerServer(
+      RssProtos.PartitionToShuffleServerRequest request,
+      StreamObserver<RssProtos.PartitionToShuffleServerResponse> responseObserver) {
+    RssProtos.PartitionToShuffleServerResponse reply;
+    RssProtos.StatusCode code;
+    int shuffleId = request.getShuffleId();
+    ShuffleHandleInfo shuffleHandleInfoByShuffleId =
+        shuffleManager.getShuffleHandleInfoByShuffleId(shuffleId);
+    if (shuffleHandleInfoByShuffleId != null) {
+      code = RssProtos.StatusCode.SUCCESS;
+      Map<Integer, List<ShuffleServerInfo>> partitionToServers =
+          shuffleHandleInfoByShuffleId.getPartitionToServers();
+      Map<Integer, RssProtos.GetShuffleServerListResponse> protopartitionToServers =
+          JavaUtils.newConcurrentMap();
+      for (Map.Entry<Integer, List<ShuffleServerInfo>> integerListEntry :
+          partitionToServers.entrySet()) {
+        List<RssProtos.ShuffleServerId> shuffleServerIds =
+            ShuffleServerInfo.toProto(integerListEntry.getValue());
+        RssProtos.GetShuffleServerListResponse getShuffleServerListResponse =
+            RssProtos.GetShuffleServerListResponse.newBuilder()
+                .addAllServers(shuffleServerIds)
+                .build();
+        protopartitionToServers.put(integerListEntry.getKey(), getShuffleServerListResponse);
+      }
+      reply =
+          RssProtos.PartitionToShuffleServerResponse.newBuilder()
+              .setStatus(code)
+              .putAllPartitionToShuffleServer(protopartitionToServers)
+              .build();
```

Comment:
Should RemoteStorage be returned here? @yl09099

Reply:
> Should RemoteStorage be returned here? @yl09099

Have been added.